cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -996,12 +996,27 @@ void ComputePageStringSizes(cudf::detail::hostdevice_span<PageInfo> pages,
   cudf::detail::join_streams(streams, stream);
 
   // check for needed temp space for DELTA_BYTE_ARRAY
-  auto const need_sizes =
-    thrust::any_of(rmm::exec_policy(stream),
-                   pages.device_begin(),
-                   pages.device_end(),
-                   cuda::proclaim_return_type<bool>(
-                     [] __device__(auto& page) { return page.temp_string_size != 0; }));
+  bool const need_sizes = [&]() -> bool {
+    // reduce with bit_or on bool-iterator to check if any pages needing temp space (equivalent to
+    // thrust::any_of)
+    std::unique_ptr<scalar> d_reduce_result = cudf::reduction::detail::reduce(
```

Comment:
Same question here: Do we see a performance advantage by replacing thrust::any_of with reduction with bit_or?